main.go

```diff
@@ -1,21 +1,38 @@
 package main
 
 import (
+	"encoding/json"
 	"flag"
 	"io/ioutil"
 	"log"
+	"os"
 )
 
 func main() {
-	configFile := flag.String("conf", "", "Config json file")
+	var config *map[string]interface{}
+	configFile := flag.String("conf", "", "Config json file, can be omitted if used in a pipe")
 	templateFile := flag.String("template", "", "Template file")
 	outFile := flag.String("out", "", "Output file")
 
 	// parse command lines
 	flag.Parse()
+	if *templateFile == "" {
+		log.Fatal("Missing template argument")
+	}
+	if *templateFile == "" {
+		log.Fatal("Missing out argument")
+	}
 
 	// Get config
-	config := readConfigFile(*configFile)
+	if *configFile == "" {
+		err := json.NewDecoder(os.Stdin).Decode(&config)
```

Comment:
We need to make sure, they don't block (which would end in blocking add-on startup)